core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -358,6 +358,18 @@ class TeamPolicyInternal : public Impl::PolicyTraits<Properties...> {
    */
   KOKKOS_INLINE_FUNCTION int team_size() const;
 
+  /** \brief Whether the policy has an automatically determined team size
+   */
+  inline bool auto_team_size() const;
+  /** \brief Whether the policy has an automatically determined vector length
+   */
+
+  inline bool auto_vector_length() const;
+
+  static int vector_length_max();
+
+  KOKKOS_INLINE_FUNCTION int vector_length() const;
```

Comment:
This wasn't here before, and while CUDA (unfortunately) exposed this as a public function the other backends didn't. If we don't have a VERY good reason to expose this to the user I would rather name this (as well as the auto_... functions impl_something. I mean the issue is that vector_length for OpenMP/Serial is at best confusing. What does it mean that vector_length on OpenMP would be 1? Does it mean we guarantee there will be no concurrency for Threadvector loops? So that is the reason why I rather would not have this function as a public API.

Reply:
It's also in our documentation as a thing TeamPolicy has, so we should update the docs to reflect what we decide https://github.com/kokkos/kokkos/wiki/Kokkos::TeamPolicy